backend/src/main/java/org/eclipse/tractusx/puris/backend/delivery/logic/adapter/DeliveryInformationSammMapper.java

```diff
@@ -0,0 +1,168 @@
+            return null;
+        }
+        Material material = deliveryList.get(0).getMaterial();
+        if (deliveryList.stream().anyMatch(deli -> !deli.getMaterial().equals(material))) {
+            log.warn("Can't map delivery list with different materials");
+            return null;
+        }
+        var groupedByPositionAttributes = deliveryList
+                .stream()
+                .collect(Collectors.groupingBy(prod -> new PositionsMappingHelper(
+                    prod.getCustomerOrderNumber(),
+                    prod.getSupplierOrderNumber(),
+                    prod.getCustomerOrderPositionNumber()
+                )));
+        DeliveryInformation samm = new DeliveryInformation();
+
+        samm.setMaterialGlobalAssetId(material.getMaterialNumberCx());
+
+        var posList = new HashSet<Position>();
+        samm.setPositions(posList);
+        for (var mappingHelperListEntry : groupedByPositionAttributes.entrySet()) {
+            var key = mappingHelperListEntry.getKey();
+            var delivery = mappingHelperListEntry.getValue().get(0);
+            Position position = new Position();
+            
+            posList.add(position);
+            if (key.customerOrderId != null || key.customerOrderPositionId != null) {
+                OrderPositionReference opr = new OrderPositionReference(
+                    delivery.getSupplierOrderNumber(),
+                    delivery.getCustomerOrderNumber(),
+                    delivery.getCustomerOrderPositionNumber()
+                );
+                position.setOrderPositionReference(opr);
+            }
+            var deliveries = new HashSet<Delivery>();
+            position.setDeliveries(deliveries);
+            for (var v : mappingHelperListEntry.getValue()) {
+                ItemQuantityEntity itemQuantityEntity = new ItemQuantityEntity(v.getQuantity(), v.getMeasurementUnit());
+                Set<TransitEvent> events = new HashSet<TransitEvent>();
+                events.add(new TransitEvent(v.getDateOfDeparture(), v.getDepartureType()));
+                events.add(new TransitEvent(v.getDateOfArrival(), v.getArrivalType()));
+                TransitLocations locations = new TransitLocations(new Location(v.getOriginBpna(), v.getOriginBpns()), new Location(v.getDestinationBpna(), v.getDestinationBpns()));
+                Delivery newDelivery = new Delivery(
+                        itemQuantityEntity, new Date(), events, locations, v.getTrackingNumber(), v.getIncoterm());
+                deliveries.add(newDelivery);
+            }
+        }
+        return samm;
+    }
+
+    public List<ReportedDelivery> sammToReportedDeliveries(DeliveryInformation samm, Partner partner) {
+        String matNbrCatenaX = samm.getMaterialGlobalAssetId();
+        ArrayList<ReportedDelivery> outputList = new ArrayList<>();
+        var mpr = mprService.findByPartnerAndPartnerCXNumber(partner, matNbrCatenaX);
```

Comment:
Same as above but here we assume we're always a customer. Please add separation of cases